bus/src/main/java/org/hawkular/metrics/component/publish/InsertedDataSubscriber.java

```diff
@@ -131,5 +149,43 @@ void shutdown() {
         if (subscription != null) {
             subscription.unsubscribe();
         }
+        publishTask.cancel();
+        publishTimer.cancel();
+    }
+
+    private int getPublishPeriod() {
+        try {
+            return Integer.parseInt(publishPeriodProperty);
+        } catch (NumberFormatException e) {
+            log.warnf("Invalid publish period. Setting default value %s", METRICS_PUBLISH_PERIOD.defaultValue());
+            return Integer.parseInt(METRICS_PUBLISH_PERIOD.defaultValue());
+        }
+    }
+
+    private class PublishTask extends TimerTask {
```

Comment:
Same here, we already have RxJava, why not use proper reactive methods here? No TimerTask is needed, just subscribe to the observable, do the filtering and publishing.

Reactive, immediate (with optional backpressure / windows if you're afraid of JMS) and fits nicely with the published format of the metrics (Observable pattern)